### PR TITLE
Implement SPV_NV_raw_access_chains

### PR DESCRIPTION
**Pull request by HansKristian-Work**
SPIRV-Headers have landed, so this can be compiled with upstream references.

dxil-spirv commit is pointing to a protected branch that compiles fine in vkd3d-proton (only headers needed), but not the full project (need spirv-tools validator, etc, and spirv-tools build is currently broken).

### CODE REVIEW

File: libs/vkd3d-shader/spirv.c [hunk 1]
```diff
@@ -10195,15 +10326,25 @@ static void vkd3d_dxbc_compiler_emit_store_uav_raw_structured(struct vkd3d_dxbc_
...
     type_id = vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_UINT, 1);
     vkd3d_dxbc_compiler_prepare_image(compiler, &image, &dst->reg, NULL, VKD3D_IMAGE_FLAG_NONE);
     assert((instruction->handler_idx == VKD3DSIH_STORE_STRUCTURED) != !image.structure_stride);
-    base_coordinate_id = vkd3d_dxbc_compiler_emit_raw_structured_addressing(compiler,
-            type_id, image.structure_stride, &src[0], VKD3DSP_WRITEMASK_0, &src[1], VKD3DSP_WRITEMASK_0);
 
-    if (image.storage_class == SpvStorageClassPhysicalStorageBuffer)
+    use_raw_access_chain = image.ssbo &&
+            vkd3d_dxbc_compiler_is_target_extension_supported(compiler,
+                    VKD3D_SHADER_TARGET_EXTENSION_RAW_ACCESS_CHAINS_NV);
```
[the remark this answers is not in the document]
Comment (HansKristian-Work) [reply libs/vkd3d-shader/spirv.c:10339]: 22.4.13 store_structured (Store Structured)

```
                    // calculate the number of components to write
                    switch (dstWriteMask)
                    {
                        x:    WriteComponents = 1; break;
                        xy:   WriteComponents = 2; break;
                        xyz:  WriteComponents = 3; break;
                        xyzw: WriteComponents = 4; break;
                        default:  // only these masks are valid
                    }
```

I don't think so, no.

**Review by doitsujin — APPROVED** [review]

**Comment by shelterx** [comment]
Anything specific needed to test this? I have
Vulkan dev 550.40.53
vkd3d-proton https://github.com/HansKristian-Work/vkd3d-proton/commit/bd1af875a7ad476271fb7d8e345c81003e1a8197 
extra/vulkan-headers 1:1.3.279-1 (vulkan-devel) [installed]
extra/vulkan-icd-loader 1.3.279-1 [installed: 1.3.276-1.1]
multilib/lib32-vulkan-icd-loader 1.3.279-1 [installed]

Tried Guardians of the Galaxy, Control and AC Valhalla and noticed no difference at all.
I do not know what kind of shaders these games use tho'.

**Comment by mbriar** [comment]
Also requires updated winevulkan that's currently only in wine git and no proton version yet.

**Comment by shelterx** [comment]
Ah, that explains it. Thank you for the quick response.

**Pull request merged**